internal/charmstore/store.go

```diff
@@ -39,30 +46,47 @@ type Pool struct {
 	Bakery     *bakery.Service
 	stats      stats
 	statsCache *cache.Cache
+	config     ServerParams
+
+	// reqStoreC is a buffered channel that contains allocated
+	// stores that are not currently in use.
+	reqStoreC chan *Store
+
+	// mu guards the fields following it.
+	mu sync.Mutex
+
+	// storeCount holds the number of stores currently allocated.
+	storeCount int
+
+	// closed holds whether the handler has been closed.
+	closed bool
 }
 
+const reqStoreCacheSize = 50
```

Comment:
Docstring? This is used when MaxMgoSessions is unspecified.
I sit really a "cache size"?